source/uwp/Renderer/lib/TileControl.cpp

```diff
@@ -0,0 +1,319 @@
+                                                             [&](IInspectable* /*sender*/, IRoutedEventArgs * /*args*/) -> HRESULT {
+                                                                 ComPtr<IUIElement> uiElement;
+                                                                 THROW_IF_FAILED(get_ResolvedImage(&uiElement));
+
+                                                                 // Extract BitmapSource from Image
+                                                                 ComPtr<IImage> image;
+                                                                 THROW_IF_FAILED(uiElement.As(&image));
+                                                                 ComPtr<IImageSource> imageSource;
+                                                                 THROW_IF_FAILED(image->get_Source(&imageSource));
+                                                                 ComPtr<IBitmapSource> bitmapSource;
+                                                                 THROW_IF_FAILED(imageSource.As(&bitmapSource));
+
+                                                                 // Extract Size from Image
+                                                                 INT32 height, width;
+                                                                 THROW_IF_FAILED(bitmapSource->get_PixelHeight(&height));
+                                                                 THROW_IF_FAILED(bitmapSource->get_PixelWidth(&width));
+
+                                                                 // Save size to member variable
+                                                                 Size imageSize;
+                                                                 imageSize.Height = (FLOAT)height;
+                                                                 imageSize.Width = (FLOAT)width;
+                                                                 THROW_IF_FAILED(put_ImageSize(imageSize));
+
+                                                                 RefreshContainerTile();
+                                                                 return S_OK;
+                                                             })
+                                                             .Get(),
+                                                         &eventToken));
+
+            THROW_IF_FAILED(image->get_Source(&imageSource));
+            THROW_IF_FAILED(m_brushXaml->put_ImageSource(imageSource.Get()));
+
+            return S_OK;
+        }
+        return E_INVALIDARG;
+    }
+
+    HRESULT TileControl::OnApplyTemplate()
+    {
+        // Call OnApplyTemplate() for composable base
+        ComPtr<IFrameworkElementOverrides> base;
+        RETURN_IF_FAILED(GetComposableBase()->QueryInterface(__uuidof(IFrameworkElementOverrides),
+                                                             reinterpret_cast<void**>(base.GetAddressOf())));
```

Comment:
You can use the .As function of ComPtr to do the QI:

        ComPtr<IInspectable> composableBase = GetComposableBase();
        ComPtr<IFrameworkElementOverrides> frameworkOverrides;
        RETURN_IF_FAILED(composableBase.As(&frameworkOverrides));

        return frameworkOverrides->OnApplyTemplate();